### PR TITLE
static analysis of shell scripts

### PR DESCRIPTION
**Pull request by wkotlarski**

### CODE REVIEW

**Comment by Expander** [comment]
Very nice, many thanks! Can you please add `createmodel` to the list of scripts to test?

**Comment by wkotlarski** [comment]
I've added

- `createmodel`
- `install-sarah`
- `createaddon`
- `flexiblesusy-conf`

**Comment by Expander** [comment]
I think `flexiblesusy-config` is created by `configure`. Can you really check it in this way?

**Comment by wkotlarski** [comment]
Oh, my bad. I'll remove it then.

**Comment by Expander** [comment]
We could also check:
~~~
config/list_sarah_model_files.sh
config/install_stripped.sh
config/convert_dos_paths.sh
config/mathematica_include_paths.sh
config/remove_export_markers.sh
utils/scan-slha.sh
utils/scan-database.sh
utils/slha_sort_blocks.sh
test/test_*.sh
~~~

**Comment by wkotlarski** [comment]
I'll add

- `config/*.sh`
- `utils/*.sh`
- `test/test_*.sh`

**Comment by Expander** [comment]
Perfect!

File: .github/workflows/static-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: static analysis
+
+on:
+  push:
+    branches:
+      - master
+      - development
+        #    paths:
```
Comment (Expander) [inline .github/workflows/static-analysis.yml:8]: Can we remove these (and the following) commented lines?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -117,6 +117,7 @@ jobs:
     # comming from the FlexibleSUSY/FlexibleSUSY repo.
     # For non PR triggers we check for the name of the repo so that if someone
     # forks FS github doesn't try to run tests on their code.
+    if: false
```
Comment (Expander) [inline .github/workflows/tests.yml:120]: I guess you will remove this line before merging, right?

Comment (wkotlarski) [reply .github/workflows/tests.yml:120]: Yes.

File: .github/workflows/static-analysis.yml (hunk 1 above)
Comment (wkotlarski) [reply .github/workflows/static-analysis.yml]: I've now set it up such that it's run only if any of scripts have changed.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (Expander) [reply .github/workflows/tests.yml:120]: Ok, thanks! Your PR is fine for me then.

**Comment by wkotlarski** [comment]
I'll wait for #386 otherwise this test will fail.

File: test/test_THDMIIMSSMBCFull_approximation.sh [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
```
Comment (wkotlarski) [inline test/test_THDMIIMSSMBCFull_approximation.sh:1]: @uukhas, shouldn't this be `/bin/sh`?

Comment (uukhas) [reply test/test_THDMIIMSSMBCFull_approximation.sh:1]: For this file it does not matter. Kind of funny, that I made a decision between `sh` and `bash` based on the file `test/test_HSSUSY_SUSYHD.sh` - the **only** test file with `bash`.

Comment (wkotlarski) [reply test/test_THDMIIMSSMBCFull_approximation.sh:1]: Ok. I'll change it to sh to make it more uniform.

**Comment by uukhas** [comment]
Thank you for this change.

**Review by Expander — APPROVED** [review]

**Pull request merged**